src/classes/BDI_AdditionalObjectService.cls

```diff
@@ -0,0 +1,367 @@
+
+                            BDI_ObjectWrap objWrap = objWrapsForInsert[t];
+
+                            if (result.isSuccess()) {
+                                objWrap.di.put(UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Imported_Record_Status_Field_Name__c),System.Label.bdiCreated);
+                                objWrap.di.put(UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Imported_Record_Field_Name__c),objWrap.sObj.Id);
+                            } else { 
+                                bdi.LogBDIError(objWrap.di, result.getErrors()[0].getMessage(),UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Imported_Record_Status_Field_Name__c));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    /*******************************************************************************************************
+    * @description Determines if there is a custom mapping logic class and constructs it if there is, otherwise returns the default
+    * mapping class.
+    * @param objMapping The Data Import Object Mapping to look for a custom mapping logic class on.
+    */
+    private BDI_ObjectMappingLogic getObjectMappingLogicClass (Data_Import_Object_Mapping__mdt objMapping) {
+        BDI_ObjectMappingLogic objMappingLogicClass;
+
+        if (objMapping.Custom_Mapping_Logic_Class__c != null) {
+            Type custLogicClassType = Type.forName(UTIL_Namespace.adjustNPSPNamespace(objMapping.Custom_Mapping_Logic_Class__c));
+            objMappingLogicClass = (BDI_ObjectMappingLogic)custLogicClassType.newInstance();
+        } else {
+            objMappingLogicClass = new BDI_ObjectMappingLogic();
+        }
+        return objMappingLogicClass;
+    }
+
+
+    /*******************************************************************************************************
+    * @description Determines if the predecessor to this Data Import Object Mapping exists
+    * @param predecessor The Data Import Object Mapping for which the imported field can be checked to see if it exists.
+    */
+    private Boolean predecessorExists(BDI_ObjectWrap objWrap, Data_Import_Object_Mapping__mdt predecessor){
+        return objWrap.di.get(UTIL_Namespace.adjustNPSPNamespace(predecessor.Imported_Record_Field_Name__c)) != null;
+    }
+
+
+    /*******************************************************************************************************
+    * @description Validates the field mappings to make sure that the required ones are not null and that at least one non imported record
+    * field is populated.
+    * @param objWrap The BDI_ObjectWrap that should have its field mappings validated.
+    */
+    private Boolean validateFields(BDI_ObjectWrap objWrap){
+
+        Boolean result = false;
+        Boolean anyNonImportedRecordFieldsPopulated = false;
+        Boolean allReqFieldsPopulated = true;
```

Comment:
prefer "allRequiredFieldsPopulated" here if only because "req" can be required, requested, etc

Reply:
Updated